cli/tools/registry/unfurl.rs

```diff
@@ -120,7 +123,23 @@ impl SpecifierUnfurler {
               ))
               .ok()
             }
-            PackageJsonDepValue::Workspace(version_req) => {
+            PackageJsonDepValue::Workspace(workspace_version_req) => {
+              let version_req = match workspace_version_req {
+                PackageJsonDepWorkspaceReq::VersionReq(version_req) => {
+                  version_req
+                }
+                // TODO(bartlomieju): this is not correct, but doesn't really matter until we
+                // start supporting publishing of npm packages
```

Comment:
I think it matters here because this is for publishing. I think also JSR might error because of the `*` version requirement? Not sure. We should look into fixing this in this PR.

Reply:
I thought JSR can't publish based on `package.json` and these version constraints can only be specified in `package.json`.

Reply:
In this case, it's currently saying that it can depend on any version of the workspace package, but the original workspace specifier is constrained to either `~` or `^`